js-packages/@prestojs/viewmodel/src/fields/DateTimeField.ts

```diff
@@ -8,11 +8,20 @@ import Field from './Field';
  *
  * Invalid datetimes are treated as Null.
  */
-export default class DateTimeField extends Field<Date> {
+export default class DateTimeField extends Field<Date, string | Date> {
     parse(value: any): Date | null {
         if (Number.isNaN(Date.parse(value))) {
             return null;
         }
         return new Date(value);
     }
+    normalize(value: string | Date): Date | null {
+        if (value instanceof Date) {
+            return value;
+        }
+        if (Number.isNaN(Date.parse(value))) {
+            return null;
+        }
+        return new Date(value);
```

Comment:
This data should always be coming from an ajax call so should be ISO format?

We're not going to run into cases where the user's browser locale changes how it's interpreted?

Reply:
yes to the first part

I have no idea about the second question... 🤔

Reply:
I was thinking about the user typing a date into a freeform text field -- the date widget should take care of that normalisation shouldn't it?

Reply:
See #45. I'm going to merge this as is and we can revisit (it's already in use like this elsewhere, eg in DateTime.ts).